controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -106,46 +103,17 @@ func (w *Workload) getControlPlaneNodes(ctx context.Context) (*corev1.NodeList,
 	return nodes, nil
 }
 
-func (w *Workload) getConfigMap(ctx context.Context, configMap ctrlclient.ObjectKey) (*corev1.ConfigMap, error) {
+func (w *Workload) GetKubeadmConfig(ctx context.Context, configMapKey ctrlclient.ObjectKey) (KubeadmConfig, error) {
 	original := &corev1.ConfigMap{}
-	if err := w.Client.Get(ctx, configMap, original); err != nil {
-		return nil, errors.Wrapf(err, "error getting %s/%s configmap from target cluster", configMap.Namespace, configMap.Name)
-	}
-	return original.DeepCopy(), nil
-}
-
-// UpdateKubernetesVersionInKubeadmConfigMap updates the kubernetes version in the kubeadm config map.
-func (w *Workload) UpdateImageRepositoryInKubeadmConfigMap(ctx context.Context, imageRepository string) error {
-	configMapKey := ctrlclient.ObjectKey{Name: "kubeadm-config", Namespace: metav1.NamespaceSystem}
-	kubeadmConfigMap, err := w.getConfigMap(ctx, configMapKey)
-	if err != nil {
-		return err
-	}
-	config := &kubeadmConfig{ConfigMap: kubeadmConfigMap}
-	if err := config.UpdateImageRepository(imageRepository); err != nil {
-		return err
+	if err := w.Client.Get(ctx, configMapKey, original); err != nil {
+		return nil, errors.Wrapf(err, "error getting %s/%s configmap from target cluster", configMapKey.Namespace, configMapKey.Name)
 	}
-	if err := w.Client.Update(ctx, config.ConfigMap); err != nil {
-		return errors.Wrap(err, "error updating kubeadm ConfigMap")
-	}
-	return nil
+	return NewKubeadmConfig(original.DeepCopy()), nil
```

Comment:
I guess we can drop the deep copy here...

Reply:
Yes makes sense, since we are patching the object in the end anyway. So the point is to actually make changes to this underlying configmap anyway.